arrays/index-START.html

```diff
@@ -0,0 +1,143 @@
+    function arrayFilter(){
+      const fifteenLong = inventors.filter(function(inventor){
+      if (inventor.year >= 1500 && inventor.year <=1599){
+        return true; // keep
+      }
+      });
+
+      const fifteenShort = inventors.filter(inventor => inventor.year >= 1500 && inventor.year <=1599); // implicit return
+
+      console.table(fifteenLong);
+    }
+    // arrayFilter();
+
+    // Array.prototype.map()
+    // 2. Give us an array of the inventors first and last names
+    function arrayMap(){
+      const fullNames = inventors.map(inventor => `${inventor.first} ${inventor.last}`);
+      console.table(fullNames)
+    }
+    // arrayMap();
+
+    // Array.prototype.sort()
+    // 3. Sort the inventors by birthdate, oldest to youngest
+    function arraySort(){
+      const ordered = inventors.sort(function(a,b){
+        if (a.year > b.year){
+          return 1;
+        } else {
+          return -1;
+        }
+      });
+
+      const shortOrdered = inventors.sort((a,b) => a.year > b.year ? 1 : -1); // tenary with implicit return
+      console.table(shortOrdered);
+    }
+    // arraySort()
+
+    // Array.prototype.reduce()
+    // 4. How many years did all the inventors live all together?
+    function arrayReduce(){
+      const totalYears = inventors.reduce((total,inventor) => {
+        return total + (inventor.passed - inventor.year);
+      },0)
+      console.log(totalYears)
+    }
+    // arrayReduce()
+
+    // 5. Sort the inventors by years lived
+    function arraySort2(){
+      const oldest = inventors.sort(function(a,b){
+        const lastGuy = a.passed - a.year;
+        const nextGuy = b.passed - b.year;
+        // return lastGuy > nextGuy ? -1 : 1;
+        if (lastGuy > nextGuy){
+          return -1;
+        } else {
+          return 1;
+        }
+      });
+      console.table(oldest);
+    }
+    // arraySort2()
+
+
+    // 6. create a list of Boulevards in Paris that contain 'de' anywhere in the name
+    // https://en.wikipedia.org/wiki/Category:Boulevards_in_Paris
+
+    // const category = Array.from(document.querySelectorAll('.mw-category a'));
+    // const de = category
+    //         .map(link => link.textContent)
+    //         .filter(streetName => streetName.includes('de'))
+
+    // 7. sort Exercise
+    // Sort the people alphabetically by last name
+    function arraySort3(){
+      const alpha = people.sort((lastOne, nextOne) =>{
+        const [aLast, aFirst] = lastOne.split(', ')
+        const [bLast, bFirst] = nextOne.split(', ')
+        return aLast > bLast ? 1 : -1;
+      });
+      console.log(alpha);
+    }
+    arraySort3()
+
+    // 8. Reduce Exercise
+    // Sum up the instances of each of these
+    const data = ['car', 'car', 'truck', 'truck', 'bike', 'walk', 'car', 'van', 'bike', 'walk', 'car', 'van', 'car', 'truck' ];
+    
+    function arrayReduce2(){
+      const transportation = data.reduce(function(obj, item){
+        if (!obj[item]){
```

Comment:
Can you explain what is going on here when you do this `if (!obj[item])`?

Reply:
If an item does not exit in the object ...

Reply:
@iemeka change to `if (!obj.hasOwnProperty(item))`